File: typing/src/vo/Kv.ts
export class Kv {
    k: string;
    v: string;

    constructor(k: string, v: string) {
        this.k = k;
        this.v = v;
    }

    static fromJson(json: any): Kv {
        return new Kv(
            json.k as string,
            json.v as string
        );
    }

    toJson(): Record<string, any> {
        return {
            k: this.k,
            v: this.v
        };
    }
}

File: lib/logic/game_server/vo/kv.dart

class Kv {
  final String k;
  final String v;

  Kv({
    required this.k,
    required this.v,
  });

  Map<String, dynamic> toJson() {
    return {
      'k': k,
      'v': v,
    };
  }

  factory Kv.fromJson(Map<String, dynamic> json) {
    return Kv(
      k: json['k'] as String,
      v: json['v'] as String,
    );
  }
}
